### PR TITLE
Docs: use vela def in custom component

### PR DESCRIPTION
**Pull request by FogDong**
use vela def in custom component

/cc @wonderflow @BinaryHB0916 @hongchaodeng

### CODE REVIEW

File: docs/platform-engineers/components/custom-component.md [hunk 1]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
 
 ## Declare `ComponentDefinition`
 
-Here is a CUE based `ComponentDefinition` example which provides a abstraction for stateless workload type:
+First, generate a `ComponentDefinition` template with `vela def init`:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
First, generate `ComponentDefinition` scaffolds via `vela def init`:
```

File: docs/platform-engineers/components/custom-component.md [hunk 2]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
-        				}]
-        			}
-        		}
-        	}
-        }
+```shell
+vela def init stateless -t component -o stateless.cue
 ```
+
+The result is as follows:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]: It generates a file:

File: docs/platform-engineers/components/custom-component.md [hunk 3]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+```
+
 In detail:
 - `.spec.workload` is required to indicate the workload type of this component.
 - `.spec.schematic.cue.template` is a CUE template, specifically:
     * The `output` filed defines the template for the abstraction.
     * The `parameter` filed defines the template parameters, i.e. the configurable properties exposed in the `Application`abstraction (and JSON schema will be automatically generated based on them).
 
-Let's declare another component named `task`, i.e. an abstraction for run-to-completion workload.
+Let's edit this auto-generated custom component file:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
Edit this auto-generated custom component file:
```

File: docs/platform-engineers/components/custom-component.md [hunk 4]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+    }
+  }
+	parameter: {
+    name:  string
+    image: string
+  }
+}
+```
+
+You can use `vela def vet` to check validation:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
You can use `vela def vet` to validate the format:
```

File: docs/platform-engineers/components/custom-component.md [hunk 5]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+
+```shell
+$ vela def vet stateless.cue
+Validation succeed.
+```
+
+Let's declare another component named `task`, i.e. an abstraction for run-to-completion workload.
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
Declare another component named `task` which is an abstraction for run-to-completion workload.
```

File: docs/platform-engineers/components/custom-component.md [hunk 6]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+
+```shell
+vela def init task -t component -o task.cue
+```
+
+The result is as follows:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]: It generates a file:

File: docs/platform-engineers/components/custom-component.md [hunk 7]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+}
+
+template: {
+	output: {}
+	parameter: {}
+}
 ```
 
-Save above `ComponentDefinition` objects to files and install them to your Kubernetes cluster by `$ kubectl apply -f stateless-def.yaml task-def.yaml`
+Edit this auto-generated custom component file:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
Edit the generated component file:
```

File: docs/platform-engineers/components/custom-component.md [hunk 8]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+	parameter: {
+    count:   *1 | int
+    image:   string
+    restart: *"Never" | string
+    cmd?: [...string]
+  }
+}
+```
+
+Install above `ComponentDefinition` objects to your Kubernetes cluster:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
Apply above `ComponentDefinition` files to your Kubernetes cluster:
```

File: docs/platform-engineers/components/custom-component.md [hunk 9]
```diff
@@ -236,98 +305,104 @@ outputs: <unique-name>:
...
+          key:  string
         }
+      }
+    }]
+    cpu?: string
+  }
+}
+```
+
+Apply it to your Kubernetes cluster:
```
Comment (hongchaodeng) [inline docs/platform-engineers/components/custom-component.md]:
```suggestion
Apply to your Kubernetes cluster:
```

File: docs/platform-engineers/components/custom-component.md [hunk 10]
```diff
@@ -8,95 +8,164 @@ In this section, it will introduce how to use [CUE](https://cuelang.org/) to dec
...
+```
+
 In detail:
 - `.spec.workload` is required to indicate the workload type of this component.
 - `.spec.schematic.cue.template` is a CUE template, specifically:
     * The `output` filed defines the template for the abstraction.
     * The `parameter` filed defines the template parameters, i.e. the configurable properties exposed in the `Application`abstraction (and JSON schema will be automatically generated based on them).
 
-Let's declare another component named `task`, i.e. an abstraction for run-to-completion workload.
+Edit this auto-generated custom component file:
```
Comment (wonderflow) [inline docs/platform-engineers/components/custom-component.md]: Please declare how this auto generated

Comment (FogDong) [reply docs/platform-engineers/components/custom-component.md]: It is declared above: https://github.com/oam-dev/kubevela.io/pull/227/files/4e51ebdc1e931bba1af116455f361bdd272ba0f3#diff-47a91ec8d3a3053891899d042dc8553a22685e02d4f931f5737c98fc0aab66a3R11

Should we describe it more specifically?

Comment (wonderflow) [reply docs/platform-engineers/components/custom-component.md]: yes, you should generate it with command that can be copy-paste and work

**Review by wonderflow — APPROVED** [review]

**Pull request merged**